client/http_sender.go

```diff
@@ -125,14 +125,11 @@ func (s *httpSender) Send(_ context.Context, call Call) {
 				// the errors we'll sweep up in this net shouldn't be retried,
 				// but we can't really know for sure which.
 				log.Warningf("failed to send HTTP request or read its response: %s", t)
-				return retry.Continue, nil
-			default:
-				// Can't retry in order to recover from this error. Propagate.
-				return retry.Break, err
+				return retry.Continue, err
 			}
 		}
-		// On successful post, we're done with retry loop.
-		return retry.Break, nil
+		// If not previously handled, propagate the error (which may be nil).
```

Comment:
I generally sort of agree, but in the case of returning `(Retry.Status, error)`, I think the cognitive overhead is lower if we avoid returning a hardcoded `nil` so that such cases are always scrutinized in code review.